test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1047,6 +1048,22 @@ public void waitForDocs(final long numDocs, final BackgroundIndexer indexer) thr
         }, maxWaitTimeMs, TimeUnit.MILLISECONDS);
     }
 
+    /**
+     * Retrieves the persistent tasks with the requested task name from the given cluster state.
+     */
+    public static List<PersistentTasksCustomMetadata.PersistentTask<?>> findTasks(ClusterState clusterState, String taskName) {
+        PersistentTasksCustomMetadata tasks = clusterState.metadata().custom(PersistentTasksCustomMetadata.TYPE);
+        return tasks.tasks().stream().filter(t -> t.getTaskName().equals(taskName)).toList();
```

Comment:
As this method is a special case of the other `findTasks` method, should we call the other one instead of reimplementing it here? 
e.g. 
```
findTasks(clusterState, Set.of(taskName))
```

Reply:
I wanted to avoid creating extra collections, but on second thought it's not worth the benefit. I will change it.